File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int,int> m;
        vector<int> ans;
        for(int i=0;i<n;i++){
            
            if(m.find(target-nums[i])!=m.end()){
               
                
                ans.push_back(m[target-nums[i]]);
                ans.push_back(i);
                
                
                return ans;
            }
            m[nums[i]]=i;
        }
        
        
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.py
from bisect import bisect_left
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        #bruteforce dp
        n = len(nums)
        dp = [1]*n
        ans=0
        for i in range(n-1,-1,-1):
            for j in range(i+1,n):
                if nums[j]>nums[i]:
                    dp[i]=max(dp[i],1+dp[j])
        return max(dp)